cmake/linker/ld/target.cmake

```diff
@@ -66,6 +66,12 @@ macro(configure_linker_script linker_script_gen linker_pass_define)
     zephyr_get_include_directories_for_lang(C current_includes)
     get_property(current_defines GLOBAL PROPERTY PROPERTY_LINKER_SCRIPT_DEFINES)
 
+    if (CMAKE_C_COMPILER MATCHES ".*/cgcc")
+      set(ld_command ${REAL_CC})
+    else()
+      set(ld_command ${CMAKE_C_COMPILER})
+    endif()
+
```

Comment:
Is there any reason this cannot be just `if("${SPARSE}" STREQUAL "y")`?

Also spurious space after `if`

Reply:
I'll check this, thanks